sdk/rust/src/http.rs

```diff
@@ -28,17 +28,75 @@ pub struct Request {
     body: Vec<u8>,
 }
 
-enum HeaderValue {
+/// A header value.
+///
+/// Since header values do not have to be valid utf8, this allows for
+/// both utf8 strings and bags of bytes.
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct HeaderValue {
+    inner: HeaderValueRep,
+}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+enum HeaderValueRep {
+    /// Header value encoded as a utf8 string
     String(String),
+    /// Header value as a bag of bytes
     Bytes(Vec<u8>),
 }
 
 impl HeaderValue {
+    /// Construct a `HeaderValue` from a string
+    pub fn string(str: String) -> HeaderValue {
```

Comment:
could `HeaderValue` implement `FromStr` instead?

Reply:
`FromStr` is for fallible operations (i.e., it's what powers the `parse` function). I don't think it's appropriate here. We could implement `From<String>` and `From<Vec<u8>>`.